examples/text-generation/habana_model_adapter.py

```diff
@@ -0,0 +1,158 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# Copyright (C) 2020-2021 Habana Labs, Ltd. an Intel Company
+###############################################################################
+
+import argparse
+import logging
+import os
+import json
+
+import lm_eval.evaluator
+import lm_eval.tasks
+import torch
+import torch.nn.functional as F
+from run_generation import setup_parser
+
+from optimum.habana.utils import get_hpu_memory_stats
+
+
+os.environ.setdefault("TOKENIZERS_PARALLELISM", "false")
+logger = logging.getLogger(__name__)
+
+
+def setup_lm_eval_parser():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter, description="Evaluation script for HPU"
+    )
+    parser.add_argument(
+        "--buckets",
+        type=int,
+        nargs="+",
+        help="Input length buckets to use with static_shapes",
+        default=[16, 32, 64, 128, 189, 284],
+    )
+
+    parser.add_argument(
+        "--output_file", "-o", type=str, help="Output file with end results and runtime parameters", required=True
+    )
+    parser.add_argument(
+        "--tasks",
+        type=str,
+        nargs="+",
+        help="Tasks to run",
+        default=["hellaswag", "lambada_openai", "piqa", "winogrande"],
+    )
+    parser.add_argument("--limit_iters", type=int, help="limit examples to run that many iterations", default=None)
+    args = setup_parser(parser)
+
+    return args
+
+
+def print_results(args, results):
+    if args.local_rank == 0:
+        if args.device == "hpu":
+            mem = get_hpu_memory_stats()
+            for k, v in mem.items():
+                print("{:35} = {} GB".format(k[:-5].replace("_", " ").capitalize(), v))
+        json.dump(results, open(args.output_file, "w"), indent=2)
```

Comment:
as mentioned offline the results are dumped twice but to same file name so the first results are overwritten